Student.js

```diff
@@ -48,106 +48,42 @@ function Student(firstName, lastName, preferenceArray, grade, sessionsPerWorksho
     };
 
     /**
-     * Assigns this student to a workshop.
+     * Assigns this student to a workshop in the student's first available slot.
      *
-     * Uses the first slot that both the student and the workshop have
-     * available.
+     * @param {workshop} workshop The workshop to add the student to.
      *
-     * @param {workshop}    The workshop to add the student to.
-     *
-     * @throws an error if the student and workshop have no common slot.
+     * @throws an error if the student already has workshops in each slot.
      */
     this.assignWorkshop = function(workshop) {
-        // If the student has an open slot that matches the workshop,
-        // assign the student to a session that works for them.
-        var slotNumber = this.findAvailableSession(workshop);
-        if (slotNumber !== null) {
-            this.assignWorkshopSession(workshop, slotNumber);
-        } else {
-            throw new Error(
-                "Could not find open session slot to add student " +
-                    this.firstName +
-                    " " +
-                    this.lastName +
-                    " to workshop " +
-                    workshop.name
-            );
-        }
-    };
-
-    /**
-     * Finds an open slot in both the student and workshop.
-     *
-     * @param {Workshop}    workshop    A workshop to put this student in.
-     *
-     * @returns {int}   A viable slot, or null if none is found
-     */
-    this.findAvailableSession = function(workshop) {
-        // Check through the times this student is available.
-        for (var i = 0; i < this.assignedWorkshops.length; i++) {
-            var slot = this.assignedWorkshops[i];
-            var session = workshop.sessions[i];
-
-            if (slot === null) {
-                // Then check the corresponding session.
-                // If there's still room in the session, we're good.
-                if (!session.hasReachedQuorum()) {
-                    return i;
-                }
+        if (this.fullyAssigned()) {
+            for (var i = 0; i < this.assignedWorkshops.length; i++) {
+                Logger.log(this.assignedWorkshops[i].name);
             }
+            Logger.log(workshop.name);
```

Comment:
Are these log messages still relevant? If they are, could you prepend each of them with a descriptive string to make the logs clearer?